File: src/main/java/com/bdev/hogwarts_api/service/event_participant/EventParticipantService.java
package com.bdev.hogwarts_api.service.event_participant;

import com.bdev.hogwarts_api.data.dto.events.EventParticipant;

import java.util.List;
import java.util.Optional;

public interface EventParticipantService {
    List<EventParticipant> getAllParticipants();
    Optional<EventParticipant> getEventParticipant(long eventParticipantId);
    List<EventParticipant> getAllEventParticipants(long eventId);
    Long createParticipant(EventParticipant eventParticipant);
    void editParticipant(EventParticipant eventParticipant);
    void deleteParticipant(long eventParticipantId);
}

File: src/main/java/com/bdev/hogwarts_api/service/event_participant/EventParticipantService.kt
package com.bdev.hogwarts_api.service.event_participant

import com.bdev.hogwarts_api.data.dto.events.EventParticipant

interface EventParticipantService {
    fun getAllParticipants(): List<EventParticipant>
    fun getEventParticipant(eventParticipantId: Long): EventParticipant?
    fun getAllEventParticipants(eventId: Long): List<EventParticipant>
    fun createParticipant(eventParticipant: EventParticipant): Long?
    fun editParticipant(eventParticipant: EventParticipant)
    fun deleteParticipant(eventParticipantId: Long)
}
